src/cs/Ssh/IO/SshProtocol.cs

```diff
@@ -160,19 +163,25 @@ internal async Task<string> ReadProtocolVersionAsync(CancellationToken cancellat
 				if (line.StartsWith("SSH-", StringComparison.Ordinal))
 				{
 					this.metrics.AddMessageReceived(i + 1);
-					return line;
+					return (line, proxyIPAddress);
+				}
+				else if (line.StartsWith("PROXY ", StringComparison.Ordinal))
```

Comment:
Support for PROXY protocol should be conditional on a session configuration setting that is off by default. Otherwise if the server is not behind a proxy then a malicious client could abuse this protocol to lie about their client IP address.

Reply:
Similarly, this must only allow a _single_ PROXY line, to prevent a malicious client from providing their own PROXY line that overwrites the one from the actual proxy.

Reply:
I have done that in tunnels repository. (reading to only single line)

Reply:
The way I am planning to block users from providing their own ip addresses to override is by not accepting proxy_protocol at nginx level.

Reply:
If the proxy protocol support is enabled in config and the incoming connection does NOT include a PROXY header, should the connection be aborted?

Reply:
> The way I am planning to block users from providing their own ip addresses to override is by not accepting proxy_protocol at nginx level.

But will nginx pass it through then?

Reply:
I am aborting the connection in our service if proxy header is not set.

Reply:
> But will nginx pass it through then?
nginx will not understand or read proxy header it is sent. So, technically there will be two lines in stream (first one our nginx and 2nd one user provided/from the user who is trying to send fake address). We read the first one from our nginx and 2nd one is skipped by our ssh library